cmd/registry-experimental/cmd/generate/openapi.go

```diff
@@ -0,0 +1,226 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/log"
+	"github.com/apigee/registry/pkg/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+	"github.com/spf13/cobra"
+)
+
+func openapiCommand(ctx context.Context) *cobra.Command {
+	var specID string
+	cmd := &cobra.Command{
+		Use:   "openapi",
+		Short: "Generate an OpenAPI spec from another specificaton format",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+
+			filter, err := cmd.Flags().GetString("filter")
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get filter from flags")
+			}
+
+			client, err := connection.NewRegistryClient(ctx)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get client")
+			}
+			// Initialize task queue.
+			taskQueue, wait := core.WorkerPoolWithWarnings(ctx, 1)
+			defer wait()
+
+			// Generate tasks.
+			name := args[0]
+			if spec, err := names.ParseSpec(name); err == nil {
+				// Iterate through a collection of specs and evaluate each.
+				err = core.ListSpecs(ctx, client, spec, filter, func(spec *rpc.ApiSpec) error {
+					taskQueue <- &generateOpenAPITask{
+						client:    client,
+						specName:  spec.Name,
+						newSpecID: specID,
+					}
+					return nil
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to list specs")
+				}
+			}
+		},
+	}
+	cmd.Flags().StringVar(&specID, "spec-id", "generated", "ID to use for generated spec")
+	return cmd
+}
+
+type generateOpenAPITask struct {
+	client    connection.RegistryClient
+	specName  string
+	newSpecID string
+}
+
+func (task *generateOpenAPITask) String() string {
+	return fmt.Sprintf("generate openapi for %s", task.specName)
+}
+
+func (task *generateOpenAPITask) Run(ctx context.Context) error {
+	log.FromContext(ctx).Info(task.String())
+
+	request := &rpc.GetApiSpecRequest{
+		Name: task.specName,
+	}
+	spec, err := task.client.GetApiSpec(ctx, request)
+	if err != nil {
+		return err
+	}
+	data, err := core.GetBytesForSpec(ctx, task.client, spec)
+	if err != nil {
+		return err
+	}
+	relation := task.newSpecID
+	var openapi string
+	if core.IsOpenAPIv2(spec.GetMimeType()) || core.IsOpenAPIv3(spec.GetMimeType()) {
```

Comment:
NIT: I find case statements easier to read.

Reply:
Updated.. like that?